app/controllers/api/person_escort_records_controller.rb

```diff
@@ -11,5 +18,22 @@ def assessment_class
     def assessment_serializer
       PersonEscortRecordSerializer
     end
+
+    def create_confirmation_event
+      now = Time.zone.now.iso8601
```

Comment:
should the occurred_at/recorded_at be passed by the client to the API, rather than assumed by the API to have occurred at the same time as the PER was updated. E.g. what if there is lag in the supplier network and it takes time (minutes...days) for the event to reach us?

Reply:
hmm very interesting question. As an initial iteration I would suggest we internally set the timestamps, similar to [journeys](https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/controllers/api/journeys_controller.rb#L139) and [moves](https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/controllers/api/v2/moves_actions.rb#L164) although those might never be created by suppliers (not sure). 
If we were to implement this however there are two approaches:
1) Do not separate the confirmed at timestamp from the recorded at: allow a client to pass in the confirmed at value and we'll use that to set the recorded at/confirmed at. This will look better on the timeline and is more inline with the PER interface and doesn't surface any internal callbacks we might be doing. However we will have to change the way we set confirmed at which is currently done in the state machine
2) Separate the confirmed at timestamp from the recorded at: either by sending the recorded at and setting the confirmed at at the time internally, or sending both, and setting those respectively. However this will surface some internal callback data I'm not keen on on the interface. 

But I would like to avoid this for now, mainly for the sake of avoiding the amendment of the PER PATCH API interface, until this might become a problem. Let me know your thoughts

Reply:
sure, that sounds reasonable for now 👍